### PR TITLE
Fix can't delete rows with binary IDs

### LINKED ISSUE #839: BUG: can't delete row with `binary` IDs in MySQL
**Describe the bug**
I tried deleting a row of a table. I right-clicked it, clicked on delete row and then on the apply button. Nothing happend

**To Reproduce**
1. install [Shopware](https://www.shopware.com/) locally
1. Open a random table
2. Try to delete a row

**Expected behavior**
I expect that the row is deleted after clicking on apply.

**Screenshots / Gifs**

**Version Information (please complete the following information):**
 - OS: Windows 10
 - App Version: 2.1.4

**Additional context**
Shopware is using `binary` for their IDs and I assume that this is the problem.
I tried it with a similar software, which showed the SQL statement it will run and there I saw something like  `... WHERE id = NULL;`.
Therefore the binary is probably the problem.

Adminer and MySQL Workbench are able to delete rows with binary IDs.

### PR DESCRIPTION
**Pull request by azmy60**
Fixes #839

### CODE REVIEW

File: apps/studio/src/lib/data/tools.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const Mutators = {
    */
   genericMutator(value: any, preserveComplex: boolean = false): JsonFriendly {
     const mutate = Mutators.genericMutator
-    if (_.isBuffer(value)) return value.toString()
```
Comment (azmy60) [inline apps/studio/src/lib/data/tools.ts:45]: The return value showed incorrect representation of the real buffer. We need to use a specific encoding like hexadecimals for example.

File: apps/studio/src/components/tableview/TableTable.vue [hunk 2]
```diff
@@ -906,12 +906,15 @@ export default Vue.extend({
...
+      const column = this.table.columns.find(c => c.columnName === pkCell.getField())
+      const isPrimaryKeyBinary = column.dataType.toUpperCase().includes('BINARY')
+
       const payload = {
         table: this.table.name,
         row: row,
         schema: this.table.schema,
         pkColumn: this.primaryKey,
-        primaryKey: pkCell.getValue()
+        primaryKey: isPrimaryKeyBinary ? Buffer.from(pkCell.getValue(), 'hex') : pkCell.getValue()
```
Comment (azmy60) [inline apps/studio/src/components/tableview/TableTable.vue:917]: knex supports handling Buffer object

File: apps/studio/src/lib/data/tools.ts (hunk 1 above)
Comment (Zerotask) [reply apps/studio/src/lib/data/tools.ts:45]: Probably toString(16)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString

Comment (rathboma) [reply apps/studio/src/lib/data/tools.ts:45]: Yeah hex is the right way to do it. Although probably we need to prefix hex strings with `s/` when displayed (although not edited), this is a convention used in many UIs for hex'd binary values

**Pull request merged**